addons/website_sale/static/src/js/website_sale.editor.js

```diff
@@ -62,6 +68,43 @@ $('.oe_website_sale').on('click', '.oe_currency_value:o_editable', function (ev)
     $(ev.currentTarget).selectContent();
 });
 
+var ProductDefaultImage = Widget.extend({
+    /**
+     * @override
+     */
+    start: function () {
+        var self = this;
+        var defaultImgPath = '/website_sale/static/src/img/product_default_img.png';
+        var $image = this.$el.find('img.product_detail_img');
+
+        $image.attr('src', defaultImgPath);
+        var $anchor = $('<a>', {
+            class: 'btn btn-link btn-lg o_product_default_anchor',
+            text: _t('Click to set a picture'),
+        });
+
+        $image.load(function () {
+            $anchor.appendTo(self.$el);
+            if (defaultImgPath !== $image.attr('src')) {
+                // Remove anchor if image changed
+                $anchor.remove();
+            }
+        });
+        return this._super.apply(this, arguments);
+    },
+});
+
+editor.Class.include({
+    /**
+     * @override
+     */
+    start: function () {
+        return this._super.apply(this, arguments).then(function () {
+            websiteRootData.websiteRootRegistry.add(ProductDefaultImage, '.o_product_default_image');
+        });
+    },
+});
+
```

Comment:
Well... I think we just wanted to change the default image when there is no image... why is there any need of JS ?

Reply:
As per the spec, we've to change default image in edit mode only. We also have to maintain the text on image that is when image is added then text should be removed from that. Text should be displayed only on default image.

Reply:
I am not reading that in the specs... I will check with FGI